From 09697ca8f7acf5e48df5665f2561419a5058c72a Mon Sep 17 00:00:00 2001
From: "Ang, Chee Hong" <chee.hong.ang@intel.com>
Date: Mon, 15 Jun 2015 10:56:36 +0800
Subject: [PATCH 28/65] gpio: gpiolib: initialize the local variable in
 function 'gpiod_set_drive' due to klocwork warning.

Signed-off-by: Ang, Chee Hong <chee.hong.ang@intel.com>
Change-Id: I32427687eb6382c46f6bf907defce6be71cd0290
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/5273
Reviewed-by: Fung, Joshua <joshua.fung@intel.com>
Reviewed-by: Tan, Raymond <raymond.tan@intel.com>
Tested-by: Tan, Raymond <raymond.tan@intel.com>
Signed-off-by: Pengyu Ma <pengyu.ma@windriver.com>
---
 drivers/gpio/gpiolib.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpio/gpiolib.c b/drivers/gpio/gpiolib.c
index 8edd0cd..e60720a 100644
--- a/drivers/gpio/gpiolib.c
+++ b/drivers/gpio/gpiolib.c
@@ -1119,7 +1119,7 @@ int gpiod_set_drive(struct gpio_desc *desc, unsigned mode)
 	unsigned long		flags;
 	struct gpio_chip	*chip;
 	int			status = -EINVAL;
-	unsigned gpio;
+	unsigned gpio = ~0;
 	spin_lock_irqsave(&gpio_lock, flags);
 
 	if (!desc || !desc->chip) {
-- 
1.7.5.4

